sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyAsyncClient.java

```diff
@@ -563,9 +620,13 @@
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<Key>> getKeyWithResponse(KeyProperties keyProperties) {
-        Objects.requireNonNull(keyProperties, "The Key Properties parameter cannot be null.");
-        return withContext(context -> getKeyWithResponse(keyProperties.getName(), keyProperties.getVersion() == null ? ""
-            : keyProperties.getVersion(), context));
+        try {
+            Objects.requireNonNull(keyProperties, "The Key Properties parameter cannot be null.");
```

Comment:
Just to confirm all exceptions including null pointer will flow through Mono.error ?

Reply:
Yes